samples/bluetooth/alexa_gadget/src/gadgets/CMakeLists.txt

```diff
@@ -0,0 +1,192 @@
+    ${ALEXA_EMBEDDED_PATH}/AlexaGadgetsProtobuf/Notifications/ClearIndicator/notificationsClearIndicatorDirective.proto
+    ${ALEXA_EMBEDDED_PATH}/AlexaGadgetsProtobuf/Notifications/ClearIndicator/notificationsClearIndicatorDirectivePayload.proto
+)
+
+set(ALERTS_PROTO_FILES
+    ${ALEXA_EMBEDDED_PATH}/AlexaGadgetsProtobuf/Alerts/SetAlert/alertsSetAlertDirective.proto
+    ${ALEXA_EMBEDDED_PATH}/AlexaGadgetsProtobuf/Alerts/SetAlert/alertsSetAlertDirectivePayload.proto
+    ${ALEXA_EMBEDDED_PATH}/AlexaGadgetsProtobuf/Alerts/DeleteAlert/alertsDeleteAlertDirective.proto
+    ${ALEXA_EMBEDDED_PATH}/AlexaGadgetsProtobuf/Alerts/DeleteAlert/alertsDeleteAlertDirectivePayload.proto
+)
+
+set(STATELISTENER_PROTO_FILES
+    ${ALEXA_EMBEDDED_PATH}/AlexaGadgetsProtobuf/Alexa.Gadget.StateListener/StateUpdate/alexaGadgetStateListenerStateUpdateDirective.proto
+    ${ALEXA_EMBEDDED_PATH}/AlexaGadgetsProtobuf/Alexa.Gadget.StateListener/StateUpdate/alexaGadgetStateListenerStateUpdateDirectivePayload.proto
+)
+
+set(SPEECHDATA_PROTO_FILES
+    ${ALEXA_EMBEDDED_PATH}/AlexaGadgetsProtobuf/Alexa.Gadget.SpeechData/Speechmarks/alexaGadgetSpeechDataSpeechmarksDirective.proto
+    ${ALEXA_EMBEDDED_PATH}/AlexaGadgetsProtobuf/Alexa.Gadget.SpeechData/Speechmarks/alexaGadgetSpeechDataSpeechmarksDirectivePayload.proto
+)
+
+set(MUSICDATA_PROTO_FILES
+    ${ALEXA_EMBEDDED_PATH}/AlexaGadgetsProtobuf/Alexa.Gadget.MusicData/Tempo/alexaGadgetMusicDataTempoDirective.proto
+    ${ALEXA_EMBEDDED_PATH}/AlexaGadgetsProtobuf/Alexa.Gadget.MusicData/Tempo/alexaGadgetMusicDataTempoDirectivePayload.proto
+)
+
+set(CUSTOM_PROTO_FILES
+    ${CMAKE_CURRENT_SOURCE_DIR}/proto/custom_event.proto
+)
+
+foreach(GROUP BASE NOTIFICATIONS ALERTS STATELISTENER SPEECHDATA MUSICDATA CUSTOM)
+    # Populate list of expected .c files output from protobuf generation
+    foreach(file ${${GROUP}_PROTO_FILES})
+        get_filename_component(FILE_PATH ${file} DIRECTORY)
+        get_filename_component(FILE_BASE ${file} NAME_WE)
+        list(APPEND PROTOC_INC_DIRS "-I.")
+        list(APPEND PROTOC_INC_DIRS "-I${FILE_PATH}")
+        list(APPEND ${GROUP}_PROTO_BYPRODUCTS "${PROTO_COMPILATION_DIR}/${FILE_BASE}.pb.c")
+    endforeach()
+
+    # Populate list of compilation commands for protoc compiler.
+    unset(PROTOC_CMD_LIST)
+    foreach(file ${${GROUP}_PROTO_FILES})
+        get_filename_component(FILE_PATH ${file} DIRECTORY)
+        get_filename_component(FILE_NAME ${file} NAME)
+        set(PROTOC_ARGS "-I."
+                        "${PROTOC_INC_DIRS}"
+                        "--nanopb_out=${PROTO_COMPILATION_DIR}"
+                        "${FILE_NAME}"
+        )
+        list(APPEND PROTOC_CMD_LIST COMMAND ${CMAKE_COMMAND} -E chdir ${FILE_PATH} ${PROTOC_BIN} ${PROTOC_ARGS})
+    endforeach()
+
+    add_custom_target(
+        protobuf_generation_${GROUP}
+        COMMAND ${PROTOC_CMD_LIST}
```

Comment:
Is `COMMAND` needed here ?
The `COMMAND` is already included in `${PROTOC_CMD_LIST}` so i'm concerned this will misbehave (but haven't tested).

Could you checkup on this ?

Reply:
Works with with and without the initial "COMMAND" here. Will remove it

Reply:
Probably the reason why none of us noticed.
(I see I did the same in my quick proposal earlier)

Some CMake functions interprets the following word as value to the argument if applied after each other.
Thanks for checking up.